prov/shm/src/smr_ep.c

```diff
@@ -61,17 +61,19 @@ int smr_setname(fid_t fid, void *addr, size_t addrlen)
 int smr_getname(fid_t fid, void *addr, size_t *addrlen)
 {
 	struct smr_ep *ep;
-	int ret = 0;
 
 	ep = container_of(fid, struct smr_ep, util_ep.ep_fid.fid);
 	if (!ep->name)
 		return -FI_EADDRNOTAVAIL;
 
 	if (!addr || *addrlen == 0 ||
 	    snprintf(addr, *addrlen, "%s", ep->name) >= *addrlen)
-		ret = -FI_ETOOSMALL;
-	*addrlen = strlen(ep->name);
-	return ret;
+		return -FI_ETOOSMALL;
```

Comment:
You need to set addrlen to the required length prior to returning.  This tells the app what size it needs to allocate.

Reply:
Ahh yes, that makes more sense to why that was the way it was. Updated.